### PR TITLE
Display the original field name when editing model column metadata

### LINKED ISSUE #24824: [Models] Always display the original field name when editing metadata
If the user adds a Display Name to a field and reorders columns, it becomes very hard to know which field (both in GUI or SQL) is sourcing that field.

We should display the original name like we do in the data model:

The original name in `mono` and the display name.

More info here: https://www.notion.so/metabase/Make-model-editing-faster-and-easier-ee31edef605c486e8f9c0fbe70fa3af2

image

**Note:** we have a similar component to this one in the native query editor’s field filter mapping selector, which doesn’t display the actual name of the selected column but which displays the table it belongs to.

### PR DESCRIPTION
**Pull request by calherries**
Resolves #24824

[Notion link](https://www.notion.so/metabase/Make-model-editing-faster-and-easier-ee31edef605c486e8f9c0fbe70fa3af2#28b8ceb4917d40068c69b08a05e2e7f6)

### How to Test

1. `+ New`
2. Question
3. Sample dataset
4. Any table
5. Save question
6. Convert it to a Model
7. Visit Model page and "Edit Metadata"
8. Click on table column headers to test for all columns:

On the right sidebar,  Display Name input should now contain a subtitle on the top left, with the original name of the db column.

##### Notes

This PR adds a 'subtitle' to the `Input` component that displays the original field name of a model's column.

Captura de Pantalla 2022-08-18 a las 13 44 28

[From Gus, on the implementation](https://github.com/metabase/metabase/pull/24859#issue-1342396031):

> We add a prop, called subtitle, that is passed to the input component.
> 
> I did consider and experiment with Cal's suggestion of a dedicated widget for this.
> 
> However, we would end up with one more widget (whose "manager" has complicated logic as it is), plus a prop to Input. Or, at best, a variation on the input which would mean more cost than benefit.
>

### CODE REVIEW

**Comment by calherries** [comment]
So far I have a half-solution working that displays the original field name. But it's high-jacking the `FormInputWidget` directly, rather than creating a new one.
image

To complete the solution, you'd need to:
(1) Create a new widget that looks correct, instead of `FormInputWidget`, copying the format of [this component](https://github.com/metabase/metabase/blob/models-metadata-display-column-name/frontend/src/metabase/query_builder/components/DataSelector.jsx#L162).
(2) Make the form use the new widget [here](https://github.com/metabase/metabase/pull/24848/files#diff-7e1354f0032d60cd0615af13526a3b296847569eba1250fcbe29592a25a6674fR82)

File: frontend/src/metabase/components/form/widgets/FormInputWidget.jsx [hunk 1]
```diff
@@ -25,24 +26,28 @@ const FormInputWidget = forwardRef(function FormInputWidget(
...
-      readOnly={readOnly}
-      autoFocus={autoFocus}
-      error={field.visited && !field.active && field.error != null}
-      rightIcon={helperText && "info"}
-      rightIconTooltip={helperText}
-      tabIndex={tabIndex}
-      fullWidth
-      ref={ref}
-    />
+    <div>
```
Comment (ranquild) [inline frontend/src/metabase/components/form/widgets/FormInputWidget.jsx:34]: It's not clear why there is a `div`. Could you explain?

File: frontend/src/metabase/core/components/Input/Input.tsx [hunk 2]
```diff
@@ -31,6 +32,7 @@ export interface InputProps extends InputAttributes {
   rightIconTooltip?: ReactNode;
   onLeftIconClick?: (event: MouseEvent<HTMLButtonElement>) => void;
   onRightIconClick?: (event: MouseEvent<HTMLButtonElement>) => void;
+  subtitle?: string;
```
Comment (ranquild) [inline frontend/src/metabase/core/components/Input/Input.tsx:35]: nit: Move it before event handlers (that start with `on`)

File: frontend/src/metabase/components/form/widgets/FormInputWidget.jsx (hunk 1 above)
Comment (calherries) [reply frontend/src/metabase/components/form/widgets/FormInputWidget.jsx:34]: Good catch, that's definitely not meant to be there.

**Review by ranquild — APPROVED** [review]

**Pull request merged**